### PR TITLE
Consolidate both Gradle projects into one

### PR DESCRIPTION
**Pull request by SalvatoreT**
Combine the "example" projects into the main project. I needed to move
the main library into it's own module so the other libraries could
exist. This allows a developer to open this project in IntelliJ and
build all of the projects easily.

There's still more clean-up to do such as consolidating the Kotlin
versions to one top level declaration, but I'm trying to make the
minimal amount of changes to get this to work.

Closes https://github.com/libp2p/jvm-libp2p/issues/153
Closes https://github.com/libp2p/jvm-libp2p/issues/152

### CODE REVIEW

File: jvm-libp2p-minimal/build.gradle.kts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import java.nio.file.Files
 import java.nio.file.Paths
 
 // To publish the release artifact to CloudSmith repo run the following :
-// ./gradlew publish -PcloudsmithUser=<user> -PcloudsmithApiKey=<api-key>
+// ./gradlew :jvm-libp2p-minima:publish -PcloudsmithUser=<user> -PcloudsmithApiKey=<api-key>
```
Comment (acompton) [inline jvm-libp2p-minimal/build.gradle.kts:12]: typo, should be `:jvm-libp2p-minimal:publish`

Comment (SalvatoreT) [reply jvm-libp2p-minimal/build.gradle.kts]: Fixed

**Comment by SalvatoreT** [comment]
Ugh. I messed up the rebase because I forgot the main branch was called `develop`. 🤦

**Comment by SalvatoreT** [comment]
Hey @Nashatyrev and @ajsutton, do y'all mind giving this a 👀 when you get the chance? If it's not something y'all want, please let me know that too, and I'll close the PR.

**Comment by Nashatyrev** [comment]
Hey @SalvatoreT 

Sorry for such long ignorance!

I think this is generally a good refactoring

Though, I tried to build on my box and got the following error: 
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':examples:android-chatter:compileReleaseKotlin'.
> SDK location not found. Define location with an ANDROID_SDK_ROOT environment variable or by setting the sdk.dir path in your project's local properties file at '/home/ubuntu/jvm-libp2p-slvatore/local.properties'.
```

Is it an option to skip `:examples:android-chatter` when android SDK is missing?

**Comment by SalvatoreT** [comment]
If you comment out that project from the `settings.gradle`, it should stop being a problem.

```diff
diff --git a/settings.gradle b/settings.gradle
index 178bb1e..b1227c7 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -2,6 +2,6 @@ include ':jvm-libp2p-minimal'
 include ':examples:pinger'
 include ':examples:chatter'
 include ':examples:cli-chatter'
-include ':examples:android-chatter'
+// include ':examples:android-chatter'
 
 rootProject.name = 'jvm-libp2p-minimal'
```

I'm not sure off the top of my head of another clean solution.

**Comment by Nashatyrev** [comment]
> If you comment out that project from the settings.gradle, it should stop being a problem.

Tried to comment his line but the build still fails for another reason
I think the project should be able to build 'out of the box' without any optional dependencies.
If you were able to modify the build this way (with either commenting android specific stuff or more gracefully with checking android SDK availability) I would finally merge this PR

**Comment by SalvatoreT** [comment]
What was the other reason it failed?

How do you run the examples in this project today?

**Pull request closed without merging**